13-cqrs/1-basic/1-before/Shipment.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * This is an example of CQS - command query separation
+ *
+ * @author Nadeesha Dilruwan <nadeesha@adsdaq.eu>
+ */
+final class Shipment
+{
+    /**
+     * @var DateTime
+     */
+    private $delivery;
+
+    /**
+     * @var string
+     */
+    private $proofOfDelivery;
+
+    /**
+     * Shipment constructor.
+     * @param string $shipmentID
+     */
+    public function __construct($shipmentID)
+    {
+        // Load current state
+    }
+
+    /**
+     * This method mutates the delivery state
+     *
+     * @param DateTime $delivery
+     * @param string $proofOfDelivery
+     */
+    public function Deliver(DateTime $delivery, $proofOfDelivery)
+    {
+        if ($delivery != null)
```

Comment:
```suggestion
        if ($delivery !== null)
```
to be in the safe side :slightly_smiling_face:

Reply:
Another suggestion would be to `Assert::null()` in the after one, haven't looked at it yet